### PR TITLE
 [TOOLS-657] Follow-up cleaning

### PR DESCRIPTION
**Pull request by lohandus**

### CODE REVIEW

File: src/main/java/objective/taskboard/controller/FollowUpProgressController.java [hunk 1]
```diff
@@ -78,38 +78,33 @@ public void initCache() {
...
-        ProgressData progressData = calculator.calculate(snapshot, project.getStartDate(), project.getDeliveryDate(), projectionTimespan);
-        return ResponseEntity.ok().body(progressData);
-
+            return new ResponseEntity<>("The project " + projectKey + " has no start date.", INTERNAL_SERVER_ERROR);
+        
+        try {
+            ProgressData progressData = followUpFacade.calculateProgressData(timezone, projectKey, project.getStartDate(), project.getDeliveryDate(), projectionTimespan);
+            return ResponseEntity.ok().body(progressData);
+            
+        } catch (ClusterNotConfiguredException e) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/controller/FollowUpProgressController.java]: MAJOR Either log or rethrow this exception. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/objective/taskboard/controller/FollowUpPlannedBallparkController.java [hunk 2]
```diff
@@ -37,14 +36,13 @@
...
-        
-        snapshot.forEachRow(accumulator::accumulate);
-
-        return ResponseEntity.ok(accumulator.getData());
+
+        try {
+            List<PlannedVsBallparkChartData> data = followUpFacade.calculatePlannedVsBallparkData(projectKey);
+            return ResponseEntity.ok(data);
+
+        } catch (ClusterNotConfiguredException e) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/controller/FollowUpPlannedBallparkController.java]: MAJOR Either log or rethrow this exception. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1166)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/followup/FollowUpDataMiner.java [hunk 3]
```diff
@@ -35,61 +12,51 @@
...
+            FollowupClusterProvider clusterProvider) {
+        this.jiraProperties = jiraProperties;
+        this.metadataService = metadataService;
+        this.issueBufferService = issueBufferService;
+        this.clusterProvider = clusterProvider;
+    }
 
-    @Autowired
-    private ProjectFilterConfigurationCachedRepository projectRepository;
+    public FollowupData getFromCurrentState(ZoneId timezone, FollowupCluster cluster, String projectKey) {
```
Comment (taksan) [inline src/main/java/objective/taskboard/followup/FollowUpDataMiner.java]: build

File: src/main/java/objective/taskboard/followup/FollowUpScopeByTypeDataProvider.java [hunk 4]
```diff
@@ -62,9 +62,9 @@ else if (rowService.isBaselineDone(r.rowData))
                 sum(map, BASELINE_DONE, effortEstimate);
             else if (rowService.isBaselineBacklog(r.rowData))
                 sum(map, BASELINE_BACKLOG, effortEstimate);
-        });
+        };
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/followup/FollowUpScopeByTypeDataProvider.java]: MINOR Remove this empty statement. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**